htdocs/accountancy/bookkeeping/card.php

```diff
@@ -198,6 +198,11 @@
 			setEventMessages($langs->trans('ErrorFieldRequired', $langs->transnoentitiesnoconv("AccountAccountingShort")), null, 'errors');
 			$action = '';
 		}
+		$subledger_account_str = is_array($subledger_account) ? reset($subledger_account) : (string) $subledger_account;
+		if (!checkGeneralAccountAllowsAuxiliary($db, $accountingaccount_number, $subledger_account)) {
```

Comment:
May be you want to use $subledger_account_str instead of $subledger_account here ?